fury/trees.py

```diff
@@ -0,0 +1,817 @@
+                self._DIVIDED = True
+
+                self.subBranches = {0 : self._DOWNLEFT,
+                                    1 : self._DOWNRIGHT,
+                                    2 : self._UPLEFT,
+                                    3 : self._UPRIGHT}
+
+                list = self.GetPointsList()
+                for i in range(self.GetPointsNumber()):
+
+                    self.SubAddPoint(list[i])
+                    self.RemovePoint(0)
+
+                self.SubAddPoint(point)
+
+            else:
+                self._POINTS = np.append(self._POINTS, point)
+                self._N_POINTS += 1
+
+    def ProcessBranch(self, Function, *args):
+        '''Recursively process the branch or its subbranches with a given function and its arguments.
+           If the function returns any value, it will be returned as the value itself or a list of the
+           values returned by each subbranch processed, if the branch is already divided.
+           * Function : Any function that has only a branch as input
+           * *args : arguments of the function in the order requested by the funciton passed'''
+        if self.IsDivided() == True:
+            list = np.array([])
+
+            for i in range(len(self.subBranches)):
+                list = np.append(
+                    list, self.GetSubBranch(i).ProcessBranch(
+                        Function, *args))
+
+            return list
+
+        else:
+            return Function(self, *args)
+
+
+class Tree2d():
+    '''Class that stores the root branch and general 2d tree informations.
+       * branch : Branch2d to be the root of the tree.'''
+
+    def __init__(self, branch : Branch2d):
+        self._ROOT = branch
+        self.xmin = branch.GetXSize()[0]
+        self.xmax = branch.GetXSize()[1]
+        self.ymin = branch.GetYSize()[0]
+        self.ymax = branch.GetYSize()[1]
+        self._N_POINTS = branch.GetTotalPoints()
+        self._STR = ""
+
+    def __str__(self):
+        if self._ROOT.IsDivided() == True:
+            points_list = []
+            for i in range(len(self.GetRoot().subBranches)):
+                points_list.append(self.GetRoot().GetSubBranch(i).GetTotalPoints())
+            length = [len(str(x)) for x in points_list]
+            length = [np.max([length[j], length[j + 2]])
+                      for j in range(len(points_list)//2)]
+            self._STR = f"number of points = {self._ROOT.GetTotalPoints()}\n"
+            self._STR += f"[ {self._ROOT._UPLEFT.GetTotalPoints()} | {self._ROOT._UPRIGHT.GetTotalPoints()} ]\n"
+            self._STR += "[---" + "-"*(length[0] - 1) + \
+                "|---" + "-"*(length[1] - 1) + f"]\n"
+            self._STR += f"[ {self._ROOT._DOWNLEFT.GetTotalPoints()} | {self._ROOT._DOWNRIGHT.GetTotalPoints()} ]\n"
+        else:
+            self._STR = f"number of points = {self._N_POINTS}\n[       ]\n[       ]\n[       ]\n"
+
+        return self._STR
+
+    def GetRoot(self):
```

Comment:
Avoid methods starting with get or set unless they are really needed . We recommend to use properties or just the name of the property.

Reply:
Ok, I will be looking into that. Is there any other place you think it didn't need to be used, or just that specific part?

Reply:
Also, adhere to the Python naming convention used in FURY: `remove_point()` instead of `RemovePoint()`.